src/utils/network/session-core.js

```diff
@@ -285,89 +144,6 @@ export const checkContainerPermission = async (session, otherPodUsername) => {
   }
 };
 
-/**
- * Function that deletes all files from a Solid container associated to a file
- * type, if exist, and returns the container's URL
- *
- * @memberof utils
- * @function deleteDocuments
- * @param {Session} session - Solid's Session Object (see {@link Session})
- * @param {string} fileType - Type of document
- * @returns {Promise} Promise - Deletes all existing files within a certain
- * container
- */
-
-export const deleteDocumentFile = async (session, fileType) => {
-  const containerToDeletedUrl = getContainerUrl(session, 'Documents', INTERACTION_TYPES.SELF);
-  const documentUrl = `${containerToDeletedUrl}${fileType.replace("'", '').replace(' ', '_')}/`;
-
-  const fetched = await getSolidDataset(documentUrl, { fetch: session.fetch });
-
-  // Solid requires all files within Pod container must be deleted before
-  // the container itself can be deleted from Pod
-  const files = getAllFiles(fetched);
-  files.filter(async (file) => {
-    if (!file.url.endsWith('/')) {
-      await deleteFile(file.url, { fetch: session.fetch });
-    }
-  });
-};
-
-/**
- * Function that generates the Documents container for users
- *
- * @memberof utils
- * @param {Session} session - Solid's Session Object
- * @param {URL} podUrl - The user's Pod URL
- * @returns {Promise} Promise - Creates Documents container for storage of
- * documents being uploaded by authorized users
- */
-
-export const createDocumentContainer = async (session, podUrl) => {
```

Comment:
We create the containers with `saveSolidDatasetAt`. That will create all necessary containers along the path, and will avoid creating unused containers. We can inherit permissions from the defaults set in either the root `.acl` or the PASS container `.acl`.

Reply:
Just saw the containers being generated along the path when testing out he upload, so this looks good to me.

Reply:
Had just tested the cross pod upload, we might want to keep this function after all, but leave it such that it'll only generate the base Documents container.

https://github.com/codeforpdx/PASS/assets/14917816/c44a0d20-e45d-4e09-9215-ff2efbf6e51c

The `setDocContainerAclPermission` would need an anchor point for the user to given permissions to other Pods. We can simply generate the Documents container with the initial ACL set for the user and the `container.ttl`.

So we can essentially delete everything after the first try/catch block for `createDocumentContainer` removing everything related to createContainerList.

For `setDocContainerAclPermission`, it'll be simplified to just:

``` JavaScript
export const setDocContainerAclPermission = async (session, permissions, otherPodUsername) => {
  const containerUrl = getContainerUrl(session, 'Documents', INTERACTION_TYPES.SELF);
  const otherPodUrl = getPodUrl(otherPodUsername);
  const webId = `${otherPodUrl}profile/card#me`;
  await setDocAclForUser(session, containerUrl, 'update', webId, permissions);
};
```

I've tried out the changes on my computer and it works well with the new functions created in this PR.

https://github.com/codeforpdx/PASS/assets/14917816/ee697b31-514c-4fe7-a946-3b3d8602882b

https://github.com/codeforpdx/PASS/assets/14917816/0aacffc4-cd9b-4201-95df-af21299b7859

Reply:
> The setDocContainerAclPermission would need an anchor point for the user to given permissions to other Pods. We can simply generate the Documents container with the initial ACL set for the user and the container.ttl.

Can we use the `PASS` container as that anchorpoint? Then just have any container underneath it inherit those permissions by default?  Then we only need to call the ACL setup function once, when the user registers, and we never have to think about it again. If that ACL changes, it's because the user intended to change it. Then if the client wants to revoke our permission to access their pod, they can just delete the one ACL. If they want to revoke permission to a sub container, they can use a different flow to add an ACL that overrides the default set at the PASS level.

Reply:
I don't think we need to make document upload worry about permissions. Document upload already has enough to think about. I think if upload fails because the user is unauthorized, we can just make the user go to a different flow to get authorization, then try to upload again.

Reply:
Think it should be just limited to Documents. We don't want other users to have full access to PASS, just for the original user.

Reply:
I don't understand what you mean. I agree the only people who should have access to a PASS container and its sub containers are the ones authorized by the owner of the pod. That's what's demonstrated in your video. The user you were uploading with didn't have access to that pod, so the upload what rejected.

I think you're talking about the question: how do we give the case manager permission to upload to the client's pod?

My point is that we can separate this from the question: how does the case manager actually upload to the client's pod?

This PR is about the latter. I think we can solve the former. If you go to your client pod, go to the PASS container, and upload an ACL that gives your case manager read-write permissions to everything under that container, then this code should work.

Reply:
> If you go to your client pod, go to the PASS container, and upload an ACL that gives your case manager read-write permissions to everything under that container, then this code should work.

Would agree that setting ACL settings would be ideal in the Documents container rather than in the PASS container. It's similar to how Credentials are set up previously.

The main difference between Documents and Credentials should be that we don't have a `setDocContainerAclPermission` equivalent for Credentials to make changes to that ACL.

Reply:
We can checkout a branch from this one and make related to permissions?